src/components/Onboarding/TimezonePicker/TimezonePicker.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {memo, useCallback, useEffect, useMemo, useState} from 'react';
+import DropDownPicker, {ValueType} from 'react-native-dropdown-picker';
+import {ActivityIndicator, Text, View} from 'react-native';
+import {styles} from './TimezonePicker.styles';
+import {colors} from '../../../constants/colors';
+import {useQuery} from '@tanstack/react-query';
+import getTimezoneOptions from '../../../api/getTimezoneOptions';
+
+function TimezonePicker() {
+  const userTz = useMemo(
+    () => Intl.DateTimeFormat().resolvedOptions().timeZone,
+    [],
+  );
+  const [isOpen, setIsOpen] = useState(false);
+  const [currentTzValue, setCurrentTzValue] = useState<ValueType | null>(null);
+  const {isFetching, isLoading, data} = useQuery({
+    queryKey: ['timezone-options'],
+    queryFn: () => getTimezoneOptions(userTz),
+    placeholderData: {
+      success: false,
+      data: {tzOptions: [], initialValueIndex: 0},
+    },
+    staleTime: Infinity,
+  });
+
+  const TzActivityIndication = useCallback(
+    ({color, size}: {color: string; size: number}) => (
+      <ActivityIndicator color={color} size={size} />
+    ),
+    [],
+  );
+
+  const {tzItems, initialValueIndex} = useMemo(
+    () => ({
+      tzItems: data?.data?.tzOptions || [],
+      initialValueIndex: data?.data?.initialValueIndex || 0,
+    }),
+    [data?.data?.initialValueIndex, data?.data?.tzOptions],
+  );
+
+  useEffect(() => {
+    if (tzItems.length > 0) {
+      setCurrentTzValue(tzItems[initialValueIndex].value);
+    }
+  }, [initialValueIndex, tzItems]);
+
+  const loading = useMemo(
```

Comment:
No. I added it because from React Native's PoV it is important to avoid unnecessary calculations or re-renders.
In web apps using React, this doesn't matter much. However in case of React Native, due to the way it works, such things can have an impact on performance of the application.
Although these things should improve with time, as the new architecture (Hermes, JSI, Turbo modules, etc.) come into picture.
Let me check if this is having an impact on the performance

Reply:
Removed it now

Reply:
how did u check the perf? thanks for the detailed reply

Reply:
Flipper.
Here is a reference: [YouTube link for conf talk](https://www.youtube.com/watch?v=5Q3VCgKV0GE)